### PR TITLE
Add conversion from grammar.json back to javascript

### PR DESCRIPTION
**Pull request by mjambon**
This produces a more readable view of the grammar, which after transformation by ocaml-tree-sitter exists in JSON format. This is for debugging. Even though the generated javascript should work, that's not the goal. We could have used another syntax for that matter. The js grammar is about half the number of lines of the JSON grammar.

For testing, compare the files `grammar-rev.json.orig` and `grammar-rev.json` after running `make test`.

### Security

- [x] Change has no security implications (otherwise, ping the security team)

### CODE REVIEW

**Review by aryx — APPROVED** [review]
Did it help to find the issue with tree-sitter-swift?
BTW we plan to have Nat Mote starting next week, so it would be great if tree-sitter-swift can be processed by ocaml-tree-sitter so he can start working on it.

File: src/gen/lib/Indent.ml [hunk 1]
```diff
@@ -119,3 +119,15 @@ let to_string nodes =
...
   Buffer.contents buf
+
+let to_channel oc nodes =
+  let data = to_string nodes in
+  output_string oc data;
+  flush oc
+
+let to_file output_file nodes =
+  let data = to_string nodes in
+  let oc = open_out output_file in
```
[the remark this answers is not in the document]
Comment (aryx) [reply src/gen/lib/Indent.ml:130]: yeah! One of our rule!

[the remark this answers is not in the document]
Comment (aryx) [reply src/gen/lib/Indent.ml:130]: We could even provide an autofix for that!

**Comment by mjambon** [comment]
> Did it help to find the issue with tree-sitter-swift?

No, not yet.

File: src/gen/lib/Indent.ml (hunk 1 above)
[the remark this answers is not in the document]
Comment (mjambon) [reply src/gen/lib/Indent.ml]: [There is an autofix](https://github.com/returntocorp/semgrep-rules/blob/65f6fdfecdf7fb7a87acb63355a3417803f08478/ocaml/lang/portability/crlf-support.yaml#L37). I don't know why it's not offered here.

[the remark this answers is not in the document]
Comment (aryx) [reply src/gen/lib/Indent.ml]: hmm maybe we need to enable it somewhere? cc @sabrinabrogren

[the remark this answers is not in the document]
Comment (mjambon) [reply src/gen/lib/Indent.ml]: @aryx @chmccreery said autofix has to be enabled on a per-project basis and she enabled it.

**Pull request merged**